core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts

```diff
@@ -0,0 +1,242 @@
+
+    public view: EditorView;
+
+    public tippy: Instance | undefined;
+
+    public tippyOptions?: Partial<Props>;
+
+    public pluginKey: PluginKey;
+
+    public component?: ComponentRef<AIContentPromptComponent>;
+
+    private $destroy = new Subject<boolean>();
+
+    constructor(props: AIContentPromptViewProps) {
+        const { editor, element, view, tippyOptions = {}, pluginKey, component } = props;
+
+        this.editor = editor;
+        this.element = element;
+        this.view = view;
+
+        this.tippyOptions = tippyOptions;
+
+        this.element.remove();
+        this.element.style.visibility = 'visible';
+        this.pluginKey = pluginKey;
+        this.component = component;
+
+        this.component.instance.formValues.pipe(takeUntil(this.$destroy)).subscribe((data) => {
+            this.editor.commands.updateValue(data);
+        });
+
+        this.component.instance.hide.pipe(takeUntil(this.$destroy)).subscribe(() => {
+            this.editor.commands.closeAIPrompt();
+        });
+
+        this.editor.on('focus', this.focusHandler);
+
+        document.body.addEventListener('scroll', this.hanlderScroll.bind(this), true);
+
+        document.body.addEventListener('mousedown', this.handleOutsideClick, { capture: true });
+    }
+
+    update(view: EditorView, prevState?: EditorState) {
+        const next = this.pluginKey?.getState(view.state);
+        const prev = prevState ? this.pluginKey?.getState(prevState) : { open: false };
+
+        const { state } = view;
+        const { doc, selection } = state;
+
+        const { ranges } = selection;
+        const from = Math.min(...ranges.map((range) => range.$from.pos));
+        const to = Math.max(...ranges.map((range) => range.$to.pos));
+
+        if (next?.open === prev?.open) {
+            this.tippy?.popperInstance?.forceUpdate();
+
+            return;
+        }
+
+        if (next.open) {
+            this.component.instance.cleanForm();
+        }
+
+        this.createTooltip();
+
+        this.tippy?.setProps({
+            getReferenceClientRect: () => {
+                if (selection instanceof NodeSelection) {
+                    const node = view.nodeDOM(from) as HTMLElement;
+
+                    if (node) {
+                        this.node = doc.nodeAt(from);
+                        const type = this.node.type.name;
+
+                        return this.tippyRect(node, type);
+                    }
+                }
+
+                return posToDOMRect(view, from, to);
+            }
+        });
+
+        next.open ? this.show() : this.hide();
+    }
+
+    createTooltip() {
+        const { element: editorElement } = this.editor.options;
+        const editorIsAttached = !!editorElement.parentElement;
+
+        if (this.tippy || !editorIsAttached) {
+            return;
+        }
+
+        this.tippy = tippy(editorElement.parentElement, {
+            ...this.tippyOptions,
+            ...TIPPY_OPTIONS,
```

Comment:
This will override the options we pass from outside, is that what we want?

Reply:
I will adjust the order so that TIPPY_OPTIONS are spread first, followed by this.tippyOptions.

For your info, I've noticed that this pattern of spreading is used in multiple places throughout codebase (in case you didn't want that).